os/kernel/init/os_bringup.c

```diff
@@ -416,5 +427,9 @@ int os_bringup(void)
 	(void)clearenv();
 #endif
 
+#ifdef CONFIG_SOFT_LOCKUP_DETECT
+	/* start monitoring for any lockup in the system */
+	(void)start_softlockup_monitor_work();
```

Comment:
This has specific usages, not wide usages module like wqueue. Then why do you make this functionality using wqeue? I think you can make a task. Wqueue is not worse at performancen than task. If this functionality should do different work at different time, wqueue is better. But actually this isn't.

Reply:
@sunghan-chang , Earlier implementation was using a high priority task which sleeps and wake up for every 10 seconds using sleep(10) api  to check whether system is alive or not.  But I changed to workqueues as it looked quite ugly to use sleep.  workqueue offered worker function which can be scheduled repeatedly, so went ahead by using workqueue.  There is no hard rule that I have to use workqueue only.  I can achieve this even by using a task . Let me know your opinion